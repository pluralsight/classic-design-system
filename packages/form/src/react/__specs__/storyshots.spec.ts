import path from 'path'
import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  configPath: path.resolve(__dirname, '../../../.storybook'),
  framework: 'react'
})

beforeEach(() => {
  jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789)
})

afterEach(() => {
  jest.spyOn(global.Math, 'random').mockRestore()
})
