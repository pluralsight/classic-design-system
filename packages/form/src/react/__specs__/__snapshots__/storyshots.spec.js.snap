// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Form.ButtonRow | aligns left 1`] = `
<div
  data-css-1so76ss=""
>
  <div
    style={
      Object {
        "padding": "24px",
        "width": "50%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid blue",
        }
      }
    >
      <div
        data-css-15rjys7=""
      >
        <div
          data-css-4l6uky=""
        >
          <div
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            form stuff
          </div>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-1cqgl9p=""
          >
            <div
              data-css-7d5932=""
            >
              <button
                data-css-126c1n5=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Save
                </span>
              </button>
            </div>
            <div
              data-css-7d5932=""
            >
              <button
                data-css-1oaxr3i=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Cancel
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Form.ButtonRow | aligns right 1`] = `
<div
  data-css-1so76ss=""
>
  <div
    style={
      Object {
        "padding": "24px",
        "width": "50%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid blue",
        }
      }
    >
      <div
        data-css-15rjys7=""
      >
        <div
          data-css-4l6uky=""
        >
          <div
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            form stuff
          </div>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-5osg3l=""
          >
            <div
              data-css-7d5932=""
            >
              <button
                data-css-126c1n5=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Save
                </span>
              </button>
            </div>
            <div
              data-css-7d5932=""
            >
              <button
                data-css-1oaxr3i=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Cancel
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Sample Form sample 1`] = `
<div
  data-css-1so76ss=""
>
  <div
    style={
      Object {
        "padding": "24px",
        "width": "50%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid blue",
        }
      }
    >
      <div
        data-css-15rjys7=""
      >
        <div
          data-css-4l6uky=""
        >
          <h2
            data-css-10w3axr=""
            size="large"
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            Sample form
          </h2>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-w0yocm=""
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-n53zhn=""
            >
              Name
            </div>
            <div
              data-css-1r967oi=""
            >
              <div
                data-css-1xm24a9=""
              >
                <div
                  data-css-177hkuc=""
                >
                  <input
                    data-css-lzb8ea=""
                    disabled={false}
                    placeholder="What's your name?"
                    size="medium"
                  />
                </div>
              </div>
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-w0yocm=""
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-n53zhn=""
            >
              Alias
            </div>
            <div
              data-css-1r967oi=""
            >
              <div
                data-css-1xm24a9=""
              >
                <div
                  data-css-177hkuc=""
                >
                  <input
                    data-css-lzb8ea=""
                    disabled={false}
                    placeholder="Do you have a nick name?"
                    size="medium"
                  />
                </div>
              </div>
            </div>
            <div
              data-css-1de0a6a=""
            >
              *Optional
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-w0yocm=""
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-n53zhn=""
            >
              Email address
            </div>
            <div
              data-css-1r967oi=""
            >
              <div
                data-css-utxo7p=""
              >
                <div
                  data-css-177hkuc=""
                >
                  <input
                    data-css-lzb8ea=""
                    disabled={false}
                    size="medium"
                    value="steve@example.com"
                  />
                </div>
              </div>
              <div
                data-css-loo18p=""
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="warning icon"
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M2 12C2 6.486 6.486 2 12 2s10 4.486 10 10-4.486 10-10 10S2 17.514 2 12zm2 0c0 4.411 3.589 8 8 8s8-3.589 8-8-3.589-8-8-8-8 3.589-8 8zm8.5 6h-1a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v1a.5.5 0 01-.5.5zm0-4h-1a.5.5 0 01-.5-.5V7a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v6.5a.5.5 0 01-.5.5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              data-css-1de0a6a=""
            >
              Not a valid email address
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-1hniuff=""
            onKeyDown={[Function]}
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-n53zhn=""
            >
              Occupation
            </div>
            <div
              data-css-8u2q7u=""
            >
              <div
                data-css-1mjw833=""
                data-css-1xm24a9=""
              >
                <div
                  data-css-ne3jjf=""
                >
                  <button
                    data-css-8bndf1=""
                    disabled={false}
                    onClick={[Function]}
                  >
                    <span
                      aria-hidden={true}
                      data-css-e6fvyw=""
                    >
                      Select your occupation...
                    </span>
                    <span
                      data-css-5m5yd6=""
                    >
                      Select your occupation...
                    </span>
                  </button>
                  <div
                    data-css-cstzed=""
                  >
                    <div
                      data-css-otejxm=""
                    >
                      <svg
                        aria-label="caret down icon"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12.5 14L8 10h9z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            style={
              Object {
                "display": "flex",
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-480533=""
            >
              <a
                data-css-1avozw2=""
                href="http://google.com"
              >
                <span
                  data-css-1997kh1=""
                >
                  Foo
                </span>
              </a>
            </div>
            <span
              style={
                Object {
                  "marginRight": "8px",
                }
              }
            />
            <div
              data-css-480533=""
            >
              <a
                data-css-1avozw2=""
                href="http://google.com"
              >
                <span
                  data-css-1997kh1=""
                >
                  Bar
                </span>
              </a>
            </div>
            <span
              style={
                Object {
                  "marginRight": "8px",
                }
              }
            />
            <div
              data-css-480533=""
            >
              <a
                data-css-1avozw2=""
                href="http://google.com"
              >
                <span
                  data-css-1997kh1=""
                >
                  Baz
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-w0yocm=""
            onKeyDown={[Function]}
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-n53zhn=""
            >
              Choose a Date
            </div>
            <div
              data-css-1xm24a9=""
            >
              <div
                data-css-17azrqv=""
              >
                <input
                  data-css-baao50=""
                  disabled={false}
                  name="mm"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="mm"
                  style={
                    Object {
                      "width": "32px",
                    }
                  }
                  value=""
                />
                <span
                  data-css-p5zzy=""
                >
                  /
                </span>
                <input
                  data-css-baao50=""
                  disabled={false}
                  name="dd"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="dd"
                  style={
                    Object {
                      "width": "28px",
                    }
                  }
                  value=""
                />
                <span
                  data-css-p5zzy=""
                >
                  /
                </span>
                <input
                  data-css-baao50=""
                  disabled={false}
                  name="yyyy"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="yyyy"
                  style={
                    Object {
                      "width": "50px",
                    }
                  }
                  value=""
                />
                <input
                  data-css-gyrv08=""
                  disabled={false}
                  readOnly={true}
                  tabIndex="-1"
                />
                <button
                  data-css-a5xmu1=""
                  disabled={false}
                  onClick={[Function]}
                >
                  <div
                    data-css-otejxm=""
                  >
                    <svg
                      aria-label="calendar icon"
                      role="img"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 7H4v3h16V7h-4v1.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V7h-4v1.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V7zm2-2h4V3.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V5h5a1 1 0 011 1v14a1 1 0 01-1 1H3a1 1 0 01-1-1V6a1 1 0 011-1h5V3.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V5zm10 7H4v7h16v-7zM5 13h2v2H5v-2zm0 3h2v2H5v-2zm3-3h2v2H8v-2zm0 3h2v2H8v-2zm3-3h2v2h-2v-2zm0 3h2v2h-2v-2zm3-3h2v2h-2v-2zm0 3h2v2h-2v-2zm3-3h2v2h-2v-2zm0 3h2v2h-2v-2z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-kn3wr0=""
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <div
              data-css-vzrwad=""
            >
              <div
                aria-checked={true}
                data-css-1sxo0xr=""
                role="checkbox"
                tabIndex="0"
              >
                <svg
                  aria-label="Checkmark"
                  data-css-a3uikz=""
                  role="img"
                  version="1.1"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="6.89667458 13 2.62114014 8.72446556 4.12826603 7.21733967 6.89667458 9.97505938 12.871734 4 14.3788599 5.51781473"
                  />
                </svg>
              </div>
            </div>
            <input
              checked={true}
              data-css-gyrv08=""
              disabled={false}
              readOnly={true}
              style={
                Object {
                  "maxWidth": "none",
                  "width": "100%",
                }
              }
              tabIndex="-1"
              type="checkbox"
              value="someVal"
            />
            <div
              data-css-pgc406=""
            >
              Checkbox selected
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-gdknmt=""
          />
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-1a26704=""
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="radiogroup"
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
            tabIndex={0}
          >
            <div
              data-css-xp79s5=""
            >
              <label
                data-css-kn3wr0=""
              >
                <div
                  data-css-1gbdc34=""
                >
                  <div
                    data-css-480533=""
                    data-css-iwspcj=""
                  >
                    <div
                      aria-checked={false}
                      data-css-dbvyuf=""
                      onFocus={[Function]}
                      role="radio"
                      tabIndex="-1"
                    />
                  </div>
                </div>
                <input
                  checked={false}
                  data-css-gyrv08=""
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex="-1"
                  type="radio"
                  value="full"
                />
                <div
                  data-css-cgsoeo=""
                >
                  Full-time employement
                </div>
              </label>
            </div>
            <div
              data-css-xp79s5=""
            >
              <label
                data-css-kn3wr0=""
              >
                <div
                  data-css-1gbdc34=""
                >
                  <div
                    data-css-480533=""
                    data-css-iwspcj=""
                  >
                    <div
                      aria-checked={false}
                      data-css-dbvyuf=""
                      onFocus={[Function]}
                      role="radio"
                      tabIndex="-1"
                    />
                  </div>
                </div>
                <input
                  checked={false}
                  data-css-gyrv08=""
                  onClick={[Function]}
                  readOnly={true}
                  tabIndex="-1"
                  type="radio"
                  value="part"
                />
                <div
                  data-css-cgsoeo=""
                >
                  Part-time employement
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-gdknmt=""
          />
        </div>
        <div
          data-css-4l6uky=""
        >
          <button
            aria-checked={false}
            data-css-nq9iqf=""
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            role="checkbox"
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
            tabIndex={0}
          >
            <div
              data-css-480533=""
            >
              <div
                data-css-1rncajx=""
              >
                <div
                  data-css-1nqr5a6=""
                />
              </div>
            </div>
            <input
              checked={false}
              data-css-6wu9td=""
              readOnly={true}
              tabIndex="-1"
              type="checkbox"
            />
            <label
              data-css-1wrf03j=""
            >
              toggle
            </label>
          </button>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-gdknmt=""
          />
        </div>
        <div
          data-css-4l6uky=""
        >
          <label
            data-css-na8jqt=""
            style={
              Object {
                "maxWidth": "none",
                "width": "100%",
              }
            }
          >
            <div
              data-css-1j9p7fe=""
            >
              Comment
            </div>
            <div
              data-css-1ios7mb=""
            >
              <div
                data-css-1xm24a9=""
              >
                <textarea
                  data-css-12a0glg=""
                  disabled={false}
                  placeholder="Anything else to report?"
                  rows={4}
                  style={
                    Object {
                      "height": 112,
                    }
                  }
                />
              </div>
            </div>
          </label>
        </div>
        <div
          data-css-4l6uky=""
        >
          <div
            data-css-1cqgl9p=""
          >
            <div
              data-css-7d5932=""
            >
              <button
                data-css-126c1n5=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Save
                </span>
              </button>
            </div>
            <div
              data-css-7d5932=""
            >
              <button
                data-css-1oaxr3i=""
                disabled={false}
                style={Object {}}
              >
                <span
                  data-css-15b13by=""
                >
                  Cancel
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
